include/sensors/core/callback_types.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+namespace sensor_callbacks {
+using MaxMessageBuffer = std::array<uint8_t, 5>;
+
+// NOLINTNEXTLINE(cppcoreguidelines-special-member-functions)
+class SingleRegisterCallback {
+  public:
+    virtual void operator()(const MaxMessageBuffer &buffer) = 0;
```

Comment:
One of the expectations of an interface like this is that someone who needs to create an implementation will know what the methods need to do. 

I don't know when `operator()()` is called verses the other `operator`. What should happen in these functions?

Reply:
I'll add documentation